### PR TITLE
Update the title of dev-ui.adoc and fix minor typos

### LINKED ISSUE #39270: Update the title of dev-ui.adoc
### Description

Based on a conversation [in PR #39235](https://github.com/quarkusio/quarkus/pull/39235#discussion_r1516029593), we need to update the title of dev-ui.adoc and fix some incidental typos.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by rolfedh**
Fixes https://github.com/quarkusio/quarkus/issues/39270
References https://github.com/quarkusio/quarkus/pull/39235#discussion_r1516029593

### CODE REVIEW

**Comment by rolfedh** [comment]
I've updated the title of dev-ui.adoc, fixed minor typos, and made minor changes in response to the ReviewDog warning feedback.

File: docs/src/main/asciidoc/dev-ui.adoc [hunk 1]
```diff
@@ -89,32 +89,32 @@ public CardPageBuildItem pages(NonApplicationRootPathBuildItem nonApplicationRoo
...
+<4> Adding the url of the external link (in this case, we use `NonApplicationRootPathBuildItem` to create this link, as this link is under the configurable non-application path, default `/q`). Always use `NonApplicationRootPathBuildItem` if your link is available under `/q`.
 <5> You can (optionally) hint the content type of the content you are navigating to. If there is no hint, a header call will be made to determine the `MediaType`;
 <6> You can add an icon. All free font-awesome icons are available.
 
 [NOTE]
 .Note about icons
 
-If you find your icon at https://fontawesome.com/search?o=r&m=free[Font awesome], you can map as follow: Example `<i class="fa-solid fa-house"></i>` will map to `font-awesome-solid:house`, so `fa` becomes `font-awesome` and for the icon name, remove the `fa-`; 
+If you find your icon at https://fontawesome.com/search?o=r&m=free[Font awesome], you can map as follow: Example `<i class="fa-solid fa-house"></i>` will map to `font-awesome-solid:house`, so `fa` becomes `font-awesome` and for the icon name, remove the `fa-`;
```
Comment (gsmet) [inline docs/src/main/asciidoc/dev-ui.adoc:101]:
```suggestion
If you find your icon at https://fontawesome.com/search?o=r&m=free[Font awesome], you can map as follow: Example `<i class="fa-solid fa-house"></i>` will map to `font-awesome-solid:house`, so `fa` becomes `font-awesome` and for the icon name, remove the `fa-`.
```

File: docs/src/main/asciidoc/dev-ui.adoc [hunk 2]
```diff
@@ -89,32 +89,32 @@ public CardPageBuildItem pages(NonApplicationRootPathBuildItem nonApplicationRoo
...
 
 ==== Embedding external content
 
-By default, even external links will render inside (embedded) in Dev UI. In the case of HTML, the page will be rendered and any other content will be shown using https://codemirror.net/[code-mirror] to markup the media type. For example the open api schema document in `yaml` format:
+By default, even external links will render inside (embedded) in Dev UI. In the case of HTML, the page will be rendered, and any other content will be shown using https://codemirror.net/[code-mirror] to markup the media type. For example, the `openapi` schema document in `yaml` format:
```
Comment (gsmet) [inline docs/src/main/asciidoc/dev-ui.adoc:105]:
```suggestion
By default, even external links will render inside (embedded) in Dev UI. In the case of HTML, the page will be rendered, and any other content will be shown using https://codemirror.net/[code-mirror] to markup the media type. For example, the OpenAPI schema document in YAML format:
```

File: docs/src/main/asciidoc/dev-ui.adoc [hunk 3]
```diff
@@ -26,42 +26,42 @@ image::dev-ui-overview-v2.png[alt=Dev UI overview,role="center"]
...
 == Add links to an extension card
 
-=== External Links
+=== External links
 
-These are links that reference other (external from Dev UI) data. This data can be HTML pages, text or other data.
+These are links that reference other (external from Dev UI) data. This data can be HTML pages, text, or other data.
 
-A good example of this is the SmallRye OpenAPI extension that contains links to the generated openapi schema in both json and yaml format, and a link to Swagger UI:
+A good example of this is the SmallRye OpenAPI extension that contains links to the generated `openapi` schema in both JSON and YAML format, and a link to Swagger UI:
```
Comment (gsmet) [inline docs/src/main/asciidoc/dev-ui.adoc:60]:
```suggestion
A good example of this is the SmallRye OpenAPI extension that contains links to the generated OpenAPI schema in both JSON and YAML format, and a link to Swagger UI:
```

**Review by gsmet — COMMENTED** [review]
Thanks, that's much better.

I suggested a few additional tweaks.

**Review by phillip-kruger — DISMISSED** [review]
Agree with all @gsmet recommendations.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I applied my suggestions and squashed the commits. It should be ready to once CI is green.

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**